internal/app/machined/pkg/controllers/network/dns_resolve_cache.go

```diff
@@ -288,3 +295,29 @@ func newDNSRunner(cfg runnerConfig, cache *dns.Cache, logger *zap.Logger) (*dns.
 
 	return dns.NewServer(serverOpts), nil
 }
+
+type stateMapper struct {
+	state state.State
+}
+
+func (s *stateMapper) ResolveAddr(ctx context.Context, name string) (netip.Addr, bool) {
+	name = strings.TrimRight(name, ".")
+
+	list, err := safe.ReaderListAll[*cluster.Member](ctx, s.state)
+	if err != nil {
+		return netip.Addr{}, false
+	}
+
+	elem, ok := list.Find(func(res *cluster.Member) bool { return res.TypedSpec().Hostname == name })
+	if !ok {
+		return netip.Addr{}, false
+	}
+
+	result := elem.TypedSpec().Addresses
+
+	if len(result) == 0 {
+		return netip.Addr{}, false
+	}
+
+	return result[0], true
```

Comment:
why one address? we should resolve to all addresses

Reply:
Done. We now return multiple DNS answers.